geth/api/api.go

```diff
@@ -195,32 +189,10 @@ func (api *StatusAPI) JailBaseJS(js string) {
 	api.b.jailManager.BaseJS(js)
 }
 
-// TODO(oskarth): API package this stuff
+// Notify and send message
 func (api *StatusAPI) Notify(token string) string {
-	log.Debug("Notify", "token", token)
-
-	var NP fcm.NotificationPayload
-	NP.Title = "Status - new message"
-	NP.Body = "ping"
-
-	// TODO(oskarth): Experiment with this
-	data := map[string]string{
-		"msg": "Hello World1",
-		"sum": "Happy Day",
-	}
-
-	ids := []string{
-		token,
-	}
-
-	c := fcm.NewFcmClient(serverKey)
-	c.NewFcmRegIdsMsg(ids, data)
-	c.SetNotificationPayload(&NP)
-
-	_, err := c.Send()
-	if err != nil {
-		log.Error("Notify failed:", err)
-	}
+	api.b.notification.Notify(token)
```

Comment:
We cant change interface 'Messaging'. It must be exactly the same as public methods in FCMClient for mocking. We cant mock a struct, but we can mock an interface.

Nice idea about provider i`m going to make that way!

About the splitting: notify makes notification, send makes send. This is just a single responsibility.

Reply:
> We cant change interface 'Messaging'. It must be exactly the same as public methods in FCMClient for mocking.

We can't design an interface because we need a mock. We plan to support more than one provider and in such a case we would need to extend this interface or make a composition of many interfaces. 

Also, it's not clear what `Messaging` and `Notification` responsibilities are. Two interfaces that I can imagine are:
1. `Notifier` which can send notifications,
2. `NotificationProvider` which is a notification backend. In our case, `FCMClient` can be encapsulated in another `struct` which implements `NotificationProvider` interface.

> About the splitting: notify makes notification, send makes send. This is just a single responsibility.

`notify` does not imply "making notification" but rather both. If we want it to be divided then let's call them closer to: `CreateNotification(token, payload) Notification` and `Send(notification)`.

Reply:
I think i got your point. I`m going try this in PR update.